### ISSUE TITLE
Validation Loss (valid_trajectories) calculated on wrong column

### ISSUE THREAD

**Issue by StefanWiegandSSW** [Nixtla/neuralforecast#728]
### What happened + What you expected to happen

Calculation of the validation loss valid_trajectories goes wrong if columns of the training data are not in the expected order, i.e.: ['unique_id',	'ds',	'y',	'month']. If I order cols: ['unique_id',	'ds',	'month',	'y'] the validation loss is wrong (most likely calculated vs month instead of y judging from the values.)
Loss for cols ordered ['unique_id',	'ds',	'y',	'month']:
![image](https://github.com/Nixtla/neuralforecast/assets/142790101/99c7f0de-83b4-451c-83ac-e990b9818ce2)

Loss for cols ordered ['unique_id',	'ds',	'month',	'y']:
![image](https://github.com/Nixtla/neuralforecast/assets/142790101/9d67349d-e0c7-4181-9ae1-27bfc6ef59b5)

Calculations should rely on the column naming not the column position, or position should be enforced by the lib.

### Versions / Dependencies

Colab

### Reproduction script

# validation loss is wrong

%%capture
!pip install neuralforecast

from neuralforecast.utils import AirPassengersDF
import pandas as pd

df_all = AirPassengersDF
# Extract the month from the 'ds' column and add it as a new column
df_all['ds'] = pd.to_datetime(df_all['ds'])
df_all['month'] = df_all['ds'].dt.month

df_all

from neuralforecast.auto import NHITS, AutoNHITS
from neuralforecast.core import NeuralForecast
horizon = 12
models = [NHITS(h = horizon,
                input_size = 5*horizon,
                mlp_units = 3 * [[64, 64]],
                max_steps=500,
                futr_exog_list = ["month"] # comment to try without
              )
          ]

%%capture
#df_all = df_all[['unique_id',	'ds',	'y',	'month']] # expected order - works fine
df_all = df_all[['unique_id',	'ds',	'month',	'y']] # unexpected order - results in wrong validation loss
nf = NeuralForecast(models=models, freq='M')
nf.fit(df=df_all, val_size=24)

import pandas as pd
mdl = nf.models[0]#.model
train = mdl.train_trajectories
valid = mdl.valid_trajectories

import matplotlib.pyplot as plt

df_loss = pd.DataFrame(train, columns=['x', 'y'])
plt.plot(df_loss['x'], df_loss['y'])
plt.title("Training Loss")
plt.show()

df_loss = pd.DataFrame(valid, columns=['x', 'y'])
plt.plot(df_loss['x'], df_loss['y'])
plt.title("Validation Loss")
plt.show()

### Issue Severity

Low: It annoys or frustrates me.

**Comment by cchallu** [comment]
Fixed and merged to main. Thanks again for submitting the issue!
